### PR TITLE
[PT-629] Restructure documentation

### PR DESCRIPTION
**Pull request by tobiasheine**
### Description
Scope of this PR is to add documentation for missing features, to separate advanced from simple usages and to change the wording to introduce the plugin as tool to parametrise builds rather than focus on android related features.

### Considerations
- advanced use cases have been extracted to `advanced-usage.md`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,27 +40,21 @@ all the properties files you intend to reference later on:
...
     secrets {
-        file project.file('secrets.properties')
+        using project.file('secrets.properties')
     }
 }
 ```
-where `secrets.properties` is a properties file that can now be referenced
-in the build script as `buildProperties.secrets`. Entries in such file can be
-accessed via the `getAt` operator:
+where `secrets.properties` is a properties file, containing key/value pairs, that can now be referenced
```
Comment (rock3r) [inline README.md]: Maybe put an example of properties file?

File: README.md [hunk 2]
```diff
@@ -42,27 +40,21 @@ all the properties files you intend to reference later on:
...
-- `boolean enabled = buildProperties.secrets['x'].boolean`
-- `int count = buildProperties.secrets['x'].int`
-- `double rate = buildProperties.secrets['x'].double`
-- `String label = buildProperties.secrets['x'].string`
-
-It is important to note that values are lazy loaded too (via the internal closure provided in `Entry`).
-Trying to access the value of a specific property could generate an exception
-if the key is missing in the provided properties file, eg:
+It is important to note that values are lazy loaded too. Trying to access the value of a specific property 
+could generate an exception if the key is missing in the provided properties file, eg:
```
Comment (rock3r) [inline README.md]: typo: `eg:` -> `e.g.:`

File: docs/advanced-usage.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
+# Advanced configuration
+
+The plugin supports a number of advanced behaviours. For example, it can be used to consume properties 
+from various external sources, it provides fallbacks for dynamic properties and can be 
+used in Android projects to map build properties to `buildConfigField` and `resValue`.   
+
+For most features examples can be also found on our [github repo](https://github.com/novoda/gradle-build-properties-plugin/tree/master/sample).
```
Comment (rock3r) [inline docs/advanced-usage.md]: Examples for most features can also be found...

File: docs/advanced-usage.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+A fallback value for a given `Entry` via the `or()` operator can be defined as:
+
+| | Example |
+|----|----|
+|another `Entry` | `buildProperties.secrets['notThere'].or(buildProperties.secrets['fallback'])` |
+|a `Closure` | `buildProperties.secrets['notThere'].or({ Math.random() })` |
+|a value | `buildProperties.secrets['notThere'].or('fallback')` |
+
+If the whole fallback chain evaluation fails then a `CompositeException` is thrown listing all
+the causes in the chain, eg:
```
Comment (rock3r) [inline docs/advanced-usage.md]: `e.g.:`

**Comment by rock3r** [comment]
It's because it's set in the repo that further pushes dismiss reviews. I don't think it's a sensible setting /shrug

**Comment by mr-archano** [comment]
@rock3r @tobiasheine I just changed the setting about dismissing stale PRs 👍

**Pull request merged**